spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -244,15 +253,51 @@ public SparkTable alterTable(Identifier ident, TableChange... changes) throws No
 
   @Override
   public boolean dropTable(Identifier ident) {
+    return dropTableWithoutPurging(ident);
+  }
+
+  @Override
+  public boolean purgeTable(Identifier ident) {
     try {
-      return isPathIdentifier(ident) ?
-          tables.dropTable(((PathIdentifier) ident).location()) :
-          icebergCatalog.dropTable(buildIdentifier(ident));
+      Table table = load(ident).first();
+      ValidationException.check(
+          PropertyUtil.propertyAsBoolean(table.properties(), GC_ENABLED, GC_ENABLED_DEFAULT),
+          "Cannot purge table: GC is disabled (deleting files may corrupt other tables)");
+      String metadataFileLocation = ((HasTableOperations) table).operations().current().metadataFileLocation();
+
+      boolean dropped = dropTableWithoutPurging(ident);
+
+      if (dropped) {
+        try {
+          // We should check whether the metadata file is existed. Because the HadoopCatalog/HadoopTables will drop the
+          // warehouse directly and ignore the `purge` argument.
+          table.io().newInputFile(metadataFileLocation).newStream().close();
+        } catch (NotFoundException e) {
+          return true;
+        } catch (IOException e) {
+          throw new UncheckedIOException(e);
+        }
+
+        SparkActions.get()
+            .deleteReachableFiles(metadataFileLocation)
+            .io(table.io())
+            .execute();
+      }
+
+      return dropped;
     } catch (org.apache.iceberg.exceptions.NoSuchTableException e) {
       return false;
     }
   }
 
+  private boolean dropTableWithoutPurging(Identifier ident) {
+    if (isPathIdentifier(ident)) {
+      return tables.dropTable(((PathIdentifier) ident).location(), false);
```

Comment:
Nit: there seems to also be guidelines about making comment when passing booleans in  https://iceberg.apache.org/contribute/.   in "Boolean Arguments" , about this very case it seems!